File: android/titanium/src/org/appcelerator/titanium/util/TitaniumJSEvent.java
/**
 * Appcelerator Titanium Mobile
 * Copyright (c) 2009 by Appcelerator, Inc. All Rights Reserved.
 * Licensed under the terms of the Apache Public License
 * Please see the LICENSE included with this distribution for details.
 */

package org.appcelerator.titanium.util;

public class TitaniumJSEvent
{
	private int eventId;
	private String successListener;
	private String errorListener;

	public TitaniumJSEvent(int eventId, String successListener, String errorListener) {
		this.eventId = eventId;
		this.successListener = successListener;
		this.errorListener = errorListener;
	}

	public TitaniumJSEvent(int eventId, String successListener) {
		this(eventId, successListener, null);
	}

	public int getEventId() {
		return eventId;
	}

	public String getSuccessListener() {
		return successListener;
	}

	public String getErrorListener() {
		return errorListener;
	}
}

File: iphone/Classes/TitaniumJSEvent.m
/**
 * Appcelerator Titanium Mobile
 * Copyright (c) 2009 by Appcelerator, Inc. All Rights Reserved.
 * Licensed under the terms of the Apache Public License
 * Please see the LICENSE included with this distribution for details.
 */

#import "TitaniumJSEvent.h"
#import "SBJSON.h"

@implementation TitaniumJSEvent
@synthesize eventName;

- (NSString *) eventString;
{
	if((eventString == nil) && (eventDict != nil)){
		eventString = [[SBJSON stringify:eventDict] retain];
	}
	return eventString;
}

- (NSDictionary *) eventDict;
{
	if((eventDict == nil) && (eventString != nil)){
		NSError * error = nil;
		SBJSON * parser = [[SBJSON alloc] init];
		eventDict = [[parser fragmentWithString:eventString error:&error] retain];
		[parser release];
		if (error != nil) {
			NSLog(@"[ERROR] Error in Fragment(%@): %@",eventString,error);
		}
	}
	return eventDict;
}

- (void) setEventString: (NSString *) newString;
{
	if((newString == eventString) || [eventString isEqualToString:newString])return;
	[eventString release];
	eventString = [newString copy];
	[eventDict release];
	eventDict = nil;
}

- (void) setEventDict: (NSDictionary *) newDict;
{
	if((eventDict == newDict) || [eventDict isEqual:newDict])return;
	[eventString release];
	eventString = nil;
	[eventDict release];
	eventDict = [newDict copy];
}


@end
